qutebrowser/utils/resources.py

```diff
@@ -0,0 +1,148 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2014-2021 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Resources related utilities"""
+
+import os
+import os.path
+import io
+import re
+import sys
+import enum
+import json
+import datetime
+import traceback
+import functools
+import contextlib
+import posixpath
+import shlex
+import mimetypes
+import pathlib
+import ctypes
+import ctypes.util
+from typing import (Any, Callable, IO, Iterator, Optional,
+                    Sequence, Tuple, Type, Union,
+                    Iterable, TypeVar, TYPE_CHECKING)
```

Comment:
Many of those imports are unused - not sure why the linters don't detect that...